docs/examples/driver_examples/Qcodes example with AMI430.ipynb

```diff
@@ -584,6 +584,56 @@
     "The ``ramp_simultaneously`` method assumes that the given setpoint value as well as the given duration are in the same units as the units of all the three individual axis instruments. If not all axis instruments have the same units for ramp rate and field, an exception will be raised."
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**NOTE** that if ``block_during_ramp`` parameter is set to ``True``, the ramp rates of the individual magnet axes will be **restored** after the end of the ramp to their original values. If ``block_during_ramp`` parameter is set to ``False``, then initiating a simultaneous ramp will change the ramp rate alues of the individual magnet axes: they will be set to values calculated from the setpoint magnetic field vector and ``vector_ramp_rate`` or desired ramp duration. If you'd like to use the simultanous ramp with ``block_during_ramp`` parameter is set to ``False`` **AND** you'd like the ramp rates of the individual magnet axes to be restored, use the ``restore_at_exit`` context manager of those ``ramp_rate`` parameters in your measurement code, see the following example:\n",
+    "\n",
+    "```python\n",
+    "# Let's assume that these are the original values\n",
+    "# of the ramp rates of the individual axes\n",
+    "ix.ramp_rate(0.01)\n",
+    "iy.ramp_rate(0.02)\n",
+    "iz.ramp_rate(0.03)\n",
+    "\n",
+    "# And let's assume that for this measurement we\n",
+    "# don't want the ramp calls to block, but we DO\n",
+    "# want to restore the original values of the ramp\n",
+    "# rates of the inidividual axes\n",
+    "i3d.block_during_ramp(False)\n",
+    "\n",
+    "with (\n",
+    "        ix.ramp_rate.restore_at_exit(),\n",
+    "        iy.ramp_rate.restore_at_exit()\n",
+    "        iz.ramp_rate.restore_at_exit()\n",
+    "):\n",
+    "    # Refer to qcodes documentation for more details about\n",
+    "    # the ``restore_at_exit`` context manager of ``Parameter``\n",
+    "\n",
+    "    i3d.ramp_simultaneously(setpoint=FieldVector(0.5, 1.0, 0.01), duration=2)\n",
```

Comment:
Would it be less error prone to split this into two functions `ramp_simultaneously_blocking / not_blocking` that way its perhaps more clear that one restores and the other does not

Reply:
for the ramp_simultanously method, probably yes, but the ramp can also be initiated suing cartersian/spherical/etc parameters, so i'm not convinced about making blocking and nonblocking versions of those (certainly not in this PR at least)

Reply:
Well I still think having the restore of the ramp rate depend on an unrelated setting is a really bad design. Can't we come up with something better

Reply:
let's think :)

Reply:
after an offline discussion, the "something better" is now that restoration happens in the wait_while_all_axes_ramping call.